ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/MnecViolationCostEvaluator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.rao_commons;
+
+import com.farao_community.farao.commons.Unit;
+import com.farao_community.farao.data.crac_api.Cnec;
+import com.farao_community.farao.data.crac_result_extensions.CnecResultExtension;
+import com.farao_community.farao.data.crac_result_extensions.ResultVariantManager;
+import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.farao_community.farao.commons.Unit.AMPERE;
+import static com.farao_community.farao.commons.Unit.MEGAWATT;
+
+/**
+ * An evaluator that computes the virtual cost resulting from the violation of
+ * the MNEC minimum margin soft constraint
+ *
+ * @author Peter Mitri {@literal <peter.mitri at rte-france.com>}
+ */
+public class MnecViolationCostEvaluator implements CostEvaluator {
```

Comment:
In link with my comment above.

I guess there is need for :
- "elementary blocks" of CostEvaluator (CostEvaluator, with an interface similar to this one).
- "combined" CostEvaluator, which give the full vision of an objective function.

Reply:
For instance the MinMarginCombinedEvaluator (MinMarginFunction) would rely on several "elementary blocks" such as : MinMarginCostEvaluator, MnecViolationCostEvaluator, LoopFlowViolationCostEvaluator, FallBackCostEvaluator, ...

Reply:
done